### PR TITLE
SearchResultEntry: detect file results not from Files provider

### PR DESCRIPTION
**Pull request by AlvaroBrey**
Some providers like FullTextSearch give file results without the `fileId` attribute.
This refactor allows for that, otherwise they would crash the UnifiedSearch UI.

Fixes https://github.com/nextcloud/android/issues/9345

### CODE REVIEW

File: src/main/java/com/owncloud/android/lib/common/SearchResultEntry.kt [hunk 1]
```diff
@@ -54,17 +59,20 @@ data class SearchResultEntry(
...
-        val sanitizer = UrlQuerySanitizer()
-        sanitizer.allowUnregisteredParamaters = true
-        sanitizer.unregisteredParameterValueSanitizer = UrlQuerySanitizer.getAllButNulLegal()
+        val sanitizer = UrlQuerySanitizer().apply {
+            allowUnregisteredParamaters = true
+            unregisteredParameterValueSanitizer = UrlQuerySanitizer.getAllButNulLegal()
+        }
+
         sanitizer.parseUrl(resourceUrl)
-        URL(resourceUrl).query?.let { sanitizer.parseQuery(it) }
```
Comment (AlvaroBrey) [inline src/main/java/com/owncloud/android/lib/common/SearchResultEntry.kt:61]: We don't need to do this. `parseUrl` already calls `parseQuery` if present. Additionally this crashes when trying to parse paths that are not valid URLs (like files paths without the server).

**Review by tobiasKaminsky — APPROVED** [review]

**Comment by tobiasKaminsky** [comment]
> Some providers like FullTextSearch give file results without the `fileId` attribute.

If we need this, please create an issue in these repositories, so this will then work automatically at some point.
👏 for having this as a fallback 👏

**Comment by AlvaroBrey** [comment]
> If we need this, please create an issue in these repositories, so this will then work automatically at some point. clap for having this as a fallback clap

Done: https://github.com/nextcloud/files_fulltextsearch/issues/143

**Pull request merged**